auth_pass_ip_restriction/models/res_users.py

```diff
@@ -0,0 +1,15 @@
+from odoo import _, models
+from odoo.exceptions import AccessDenied
+from odoo.http import request
+from odoo.tools.config import config
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    def _check_credentials(self, password, env):
+        ip = request.httprequest.environ["REMOTE_ADDR"]
```

Comment:
Unfortunately, [tests are failing because of this line](https://github.com/OCA/server-auth/actions/runs/4286094456/jobs/7465211683?pr=489#step:8:821), because we don't always have the `request` object available; only when this actually gets called from a HTTP context.

Odoo themselves seem to be doing this:

```
ip = request.httprequest.environ['REMOTE_ADDR'] if request else 'n/a'
```

which is a reasonable approach; so then what seems logical to me is if we can't find a request object and hence no IP, we do allow the authentication request to go through, since it doesn't come from an IP so it must be some internal check.

Reply:
ok, I've added a check for `request` together with the check for `bypass_ip_check`, which should have the same effect you propose I think.

Reply:
Still some pre-commit issue